core/domain/user_jobs_one_off.py

```diff
@@ -25,6 +26,55 @@
         models.NAMES.feedback, models.NAMES.user]))
 
 
+class UserContributionsOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for creating and populating UserContributionsModels for 
+    all registered users that have contributed.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotMetadataModel,
+            user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+
+        if isinstance(item, exp_models.ExplorationSnapshotMetadataModel):
+
+            split_id = item.id.rsplit("-")
+            yield (item.committer_id, {
+                'version_number': split_id[1],
+                'exploration_id': split_id[0]
+            })
+        elif isinstance(item, user_models.UserSettingsModel):
+
+            yield(item.id, None)
+
+
+    @staticmethod
+    def reduce(key, version_and_exp_ids):
+
+        created_explorations = set()
+        edited_explorations = set()
+
+        edits = [ast.literal_eval(v) for v in version_and_exp_ids]
+
+        for edit in edits:
+            if edit is not None:
```

Comment:
I'm guessing, maybe this is an attempt to create models for users that don't already have them?

If so, I wouldn't bother, since you've added a get-or-create paradigm for adding edited/created explorations. But please let me know if I'm missing something here.